debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/OptInfo.java

```diff
@@ -293,4 +294,14 @@ public static J9UTF8Pointer getPreloadClassNameAtIndex(J9ROMClassPointer romClas
 		}
 		return J9UTF8Pointer.NULL;
 	}
+
+	public static int getImplicitCreationFlags(J9ROMClassPointer romClass) throws CorruptDataException {
+		SelfRelativePointer srpPtr = getSRPPtr(J9ROMClassHelper.optionalInfo(romClass), romClass.optionalFlags(),
+			J9_ROMCLASS_OPTINFO_IMPLICITCREATION_ATTRIBUTE);
+		if (srpPtr.notNull()) {
+			U32Pointer u32Ptr = U32Pointer.cast(srpPtr.get());
+			return u32Ptr.at(0).intValue();
+		}
+		return 0;
+	}
```

Comment:
`int` might not be a good return type for this method: if ever we have a flag using the most-significant bit, `U32.intValue()` will throw `InvalidDataTypeException`. Perhaps the return type should be `U32`? Alternatively, it could say
```java
			return (int)u32Ptr.at(0).longValue();
```

Reply:
Thanks. I updated the return type to U32.